mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp

```diff
@@ -483,6 +489,23 @@ getSizesAndStrides(int64_t rank1, StridedMemRefType<float, 4> *filter,
     outputSizeStride[outputLayout[i]] =
         std::make_pair(outputSizes[i], outputStrides[i]);
   }
+
+  // Move sizes and strides into vectors to help compiler optimization
+  // filter: k c y x
+  fSizes = {filterSizeStride['k'].first, filterSizeStride['c'].first,
+            filterSizeStride['y'].first, filterSizeStride['x'].first};
+  fStrides = {filterSizeStride['k'].second, filterSizeStride['c'].second,
+              filterSizeStride['y'].second, filterSizeStride['x'].second};
+  // input: n c h w
+  iSizes = {inputSizeStride['n'].first, inputSizeStride['c'].first,
+            inputSizeStride['h'].first, inputSizeStride['w'].first};
+  iStrides = {inputSizeStride['n'].second, inputSizeStride['c'].second,
+              inputSizeStride['h'].second, inputSizeStride['w'].second};
+  // output: n k h w
+  oSizes = {outputSizeStride['n'].first, outputSizeStride['k'].first,
+            outputSizeStride['h'].first, outputSizeStride['w'].first};
+  oStrides = {outputSizeStride['n'].second, outputSizeStride['k'].second,
+              outputSizeStride['h'].second, outputSizeStride['w'].second};
   return;
```

Comment:
This is certainly much faster but not as good as a local array that can be SROA'd. But I think the difference is probably negligible.